### PR TITLE
Handle errors with promise rejections

### PR DESCRIPTION
**Pull request by ankon**
The code used promises, but in most cases dropped errors and rejects "on the floor", rather than just
exposing them further to the caller. This made it impossible to actually catch these problems, leading
to NodeJS warnings about "unhandled promise rejections" (which will soon make the NodeJS process
crash!)

---
This is related to https://github.com/gitbrent/PptxGenJS/issues/678, and at least allows to catch the errors.

The fixes follow a couple of patterns:

1. Remove useless wrapping promises when there is already one available

   ```js
   return new Promise((resolve, _reject) => doSomePromiseCode.then((result) => resolve(result)))
   ```

   This can be written as simply `return doSomePromiseCode`, which also makes the caller able to handle the rejection.

2. Move the use of `new Promise()` calls for callbacks as close to the callback as possible.

3. Make sure to return promises (for example the result of `Promise.all`)

Note that this does still contain the `Presenation` typo in `writeFile`, see #694 for that.

### CODE REVIEW

**Comment by ankon** [comment]
I've tested this by updating our application to use a pptxgenjs distribution compiled from this PR, and seeing that we now correctly catch the errors caused by undefined `text` given to `addSlide` in the saving path (see #678 for that problem)

File: src/pptxgen.ts [hunk 1]
```diff
@@ -560,15 +539,10 @@ export default class PptxGenJS implements IPresentationLib {
...
-				.then(content => {
-					resolve(content)
-				})
-				.catch(ex => {
-					reject(ex + '\nDid you mean to use writeFile() instead?')
-				})
-		})
+		return this.exportPresentation(outputType)
+			.catch(ex => {
+				throw new Error(ex.message + '\nDid you mean to use writeFile() instead?')
```
Comment (ankon) [inline src/pptxgen.ts]: FWIW: This kills the stack trace of the actual exception, which seriously sucks for debugging. I would suggest to actually drop this as well, and just do:

```js
write(outputType: JSZIP_OUTPUT_TYPE): Promise<string | ArrayBuffer | Blob | Buffer | Uint8Array> {
	return this.exportPresentation(outputType)
}
```

**Comment by ankon** [comment]
@gitbrent do you maybe have some time to look over this? Right now the PptxGenJS library kills our processes due to the unhandled rejections, and this patch seems to fix things nicely -- but of course we don't want to run with local patches if possible :)

**Comment by ankon** [comment]
Anything I can do to help move this forward?

**Comment by ankon** [comment]
Link to #719 (fixed also by #720).

**Comment by ankon** [comment]
Rebased onto current master to resolve the conflict introduced through #694

**Comment by gitbrent** [comment]
Thanks @ankon - great work! Appreciate the Promise expertise.

**Pull request merged**